jax/random.py

```diff
@@ -120,26 +120,59 @@
+available implementations are:
 
--   **default**, `"threefry2x32"`:
-    `A counter-based PRNG built around the Threefry hash function <http://www.thesalmons.org/john/random123/papers/random123sc11.pdf>`_.
--   *experimental* A PRNG that thinly wraps the XLA Random Bit Generator (RBG) algorithm. See
-    `TF doc <https://www.tensorflow.org/xla/operation_semantics#rngbitgenerator>`_.
+-   ``"threefry2x32"`` (**default**):
+    A counter-based PRNG based on a variant of the Threefry hash function,
+    as described in `this paper by Salmon et al., 2011
+    <http://www.thesalmons.org/john/random123/papers/random123sc11.pdf>`_.
 
-    -   `"rbg"` uses ThreeFry for splitting, and XLA RBG for data generation.
-    -   `"unsafe_rbg"` exists only for demonstration purposes, using RBG both for
-        splitting (using an untested made up algorithm) and generating.
+-   ``"rbg"`` and ``"unsafe_rbg"`` (**experimental**): PRNGs built atop
+    `XLA's Random Bit Generator (RBG) algorithm
+    <https://openxla.org/xla/operation_semantics#rngbitgenerator>`_.
 
-    The random streams generated by these experimental implementations haven't
-    been subject to any empirical randomness testing (e.g. Big Crush). The
-    random bits generated may change between JAX versions.
+    - ``"rbg"`` uses XLA RBG for random number generation, whereas for
+      key derivation (as in ``jax.random.split`` and
+      ``jax.random.fold_in``) it uses the same method as
+      ``"threefry2x32"``.
 
-The possible reasons not use the default RNG are:
+    - ``"unsafe_rbg"`` uses XLA RBG for both generation as well as key
+      derivation.
 
-1.  it may be slow to compile (specifically for Google Cloud TPUs)
-2.  it's slower to execute on TPUs
-3.  it doesn't support efficient automatic sharding / partitioning
+    Random numbers generated by these experimental schemes have not
+    been subject to empirical randomness testing (e.g. BigCrush).
+
+    Key derivation in ``"unsafe_rbg"`` has also not been empirically
+    tested. The name emphasizes "unsafe" because key derivation
+    quality and generation quality are not well understood.
+
+    Additionally, both ``"rbg"`` and ``"unsafe_rbg"`` behave unusually
+    under ``jax.vmap``. When vmapping a random function over a batch
+    of keys, its output values can differ from its true map over the
+    same keys. Instead, under ``vmap``, the entire batch of output
+    random numbers is generated from only the first key in the input
+    key batch. For example, if ``keys`` is a vector of 8 keys, then
+    ``jax.vmap(jax.random.normal)(keys)`` equals
+    ``jax.random.normal(keys[0], shape=(8,))``. This peculiarity
+    reflects a workaround to XLA RBG's limited batching support.
+
+Reasons to use an alternative to the default RNG include that:
+
+1. It may be slow to compile for TPUs.
+2. It is relatively slower to execute on TPUs.
+
+**To support efficient automatic partitioning**, all implementations
```

Comment:
You might clarify the context here: this is efficient partitioning in the case of sharded computations, yes?